### ISSUE TITLE
creating MetricExporter disables OpenTelemetry if opentelemetry-opencensus-shim is in use

### ISSUE THREAD

**Issue by psx95** [GoogleCloudPlatform/opentelemetry-operations-java#202]
Very similar to  #141 and should be resolved in a similar way. Look at #141 for details.

### Summary: 
Using exporter-trace dependency with opencensus-shim causes a failure. The issue was able to be reproduced with the same error - 

```
Exception in thread "main" java.lang.IllegalStateException: GlobalOpenTelemetry.set has already been called. GlobalOpenTelemetry.set must be called only once before any calls to GlobalOpenTelemetry.get. If you are using the OpenTelemetrySdk, use OpenTelemetrySdkBuilder.buildAndRegisterGlobal instead. Previous invocation set to cause of this exception.
	at io.opentelemetry.api.GlobalOpenTelemetry.set(GlobalOpenTelemetry.java:99)
	at io.opentelemetry.sdk.OpenTelemetrySdkBuilder.buildAndRegisterGlobal(OpenTelemetrySdkBuilder.java:84)
	at com.example.springboot.Application.initializeOTel(Application.java:96)
	at com.example.springboot.Application.main(Application.java:44)
Caused by: java.lang.Throwable
	at io.opentelemetry.api.GlobalOpenTelemetry.set(GlobalOpenTelemetry.java:107)
	at io.opentelemetry.api.GlobalOpenTelemetry.get(GlobalOpenTelemetry.java:77)
	at io.opentelemetry.api.GlobalOpenTelemetry.getTracer(GlobalOpenTelemetry.java:126)
	at io.opentelemetry.opencensusshim.OpenTelemetrySpanBuilderImpl.<clinit>(OpenTelemetrySpanBuilderImpl.java:52)
	at io.opentelemetry.opencensusshim.OpenTelemetryTracerImpl.spanBuilderWithExplicitParent(OpenTelemetryTracerImpl.java:42)
	at io.opencensus.trace.Tracer.spanBuilder(Tracer.java:308)
	at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:865)
	at com.google.cloud.ServiceOptions.getAppEngineProjectIdFromMetadataServer(ServiceOptions.java:503)
	at com.google.cloud.ServiceOptions.getAppEngineProjectId(ServiceOptions.java:472)
	at com.google.cloud.ServiceOptions.getDefaultProjectId(ServiceOptions.java:379)
	at com.google.cloud.opentelemetry.metric.MetricConfiguration.<clinit>(MetricConfiguration.java:42)
	at com.google.cloud.opentelemetry.metric.GoogleCloudMetricExporter.createWithDefaultConfiguration(GoogleCloudMetricExporter.java:76)
	at com.example.springboot.Application.initializeOTel(Application.java:79)
	... 1 more

```

**Comment by dashpole** [comment]
Marking p1 since this is important for the OC migration.

**Comment by psx95** [comment]
Able to reproduce it when `GOOGLE_CLOUD_PROJECT` env variable is not set and `ServiceOptions.getDefaultProjectId()` returns a non-empty project ID. 

Fix should be similar to the deferred approach followed for exporter-trace.
